src/pages/workspaces/WorkspacesList/WorkspaceUserActions.ts

```diff
@@ -0,0 +1,24 @@
+import { createContext } from 'react';
+import { WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+export const WorkspaceUserActionsContext = createContext<WorkspaceUserActionsState>({
+  userActions: {
+    creatingNewWorkspace: false,
+  },
+  setUserActions: () => {},
+});
+
+export interface WorkspaceUserActionsState {
+  userActions: WorkspaceUserActions;
+  setUserActions: (actions: Partial<WorkspaceUserActions>) => void;
```

Comment:
I think there's an opportunity to define a better interface here by wrapping the context.

Maybe something like:
```ts
const useActioningWorkspace = (): {
  workspaceId: string;
  action: 'creating' | 'cloning' | ...
```
for the container component

and
```ts
const useWorkspaceActions = (workspaceId: string) => {
  const context = useContext(WorkspaceUserActionsContext);
  return {
    clone: () => { context.set({ cloningWorkspaceId: workspaceId }) }
    ...
```
for the individual workspace actions.